### PR TITLE
Improve ts6307 error

### LINKED ISSUE #31651: Clarify which project has missing reference
While moving my TS project to using the new references I'm encountering a ton of the following errors.

`error TS6307: File '<path>/svgsurface.ts' is not in project file list. Projects must list all files or use an 'include' pattern.`

This error would be much more helpful if it included the name of the project this originates from... (I have approx 21 projects, and 450 of these errors)....

### PR DESCRIPTION
**Pull request by paulvanbrenk**
Fixes #31651

### CODE REVIEW

File: src/compiler/program.ts [hunk 1]
```diff
@@ -2773,7 +2773,7 @@ namespace ts {
                     // Ignore file that is not emitted
                     if (!sourceFileMayBeEmitted(file, options, isSourceFileFromExternalLibrary, getResolvedProjectReferenceToRedirect)) continue;
                     if (rootPaths.indexOf(file.path) === -1) {
-                        programDiagnostics.add(createCompilerDiagnostic(Diagnostics.File_0_is_not_in_project_file_list_Projects_must_list_all_files_or_use_an_include_pattern, file.fileName));
+                        programDiagnostics.add(createCompilerDiagnostic(Diagnostics.File_0_is_not_in_project_1_file_list_Projects_must_list_all_files_or_use_an_include_pattern, file.fileName, options.configFilePath));
```
Comment (sheetalkamat) [inline src/compiler/program.ts]: Outside toolings  may not set `options.configFilePath` so better to write it as `options.config || ""` ?

Comment (paulvanbrenk) [reply src/compiler/program.ts]: Updated

**Comment by DanielRosenwasser** [comment]
Just as a heads up, your commits don't seem to be associated with your GitHub account. While this isn't technically a problem, you might care if you want more appropriate attribution. You can either make sure your GitHub account is associated with the email address you're using for your commits, or rebase and amend your commits to fix the author name and email.

**Comment by paulvanbrenk** [comment]
> Just as a heads up, your commits don't seem to be associated with your GitHub account. While this isn't technically a problem, you might care if you want more appropriate attribution. You can either make sure your GitHub account is associated with the email address you're using for your commits, or rebase and amend your commits to fix the author name and email.

That's what happens when you use your work laptop....

File: src/compiler/diagnosticMessages.json [hunk 2]
```diff
@@ -3956,7 +3956,7 @@
         "category": "Error",
         "code": 6306
     },
-    "File '{0}' is not in project file list. Projects must list all files or use an 'include' pattern.": {
+    "File '{0}' is not in project '{1}' file list. Projects must list all files or use an 'include' pattern.": {
```
Comment (DanielRosenwasser) [inline src/compiler/diagnosticMessages.json]:
```
File '{0}' is not listed within the file list of project '{1}'.
```

Comment (paulvanbrenk) [reply src/compiler/diagnosticMessages.json]: Sounds good, keeping the second half too.

**Pull request merged**